### PR TITLE
Handle known errors while submitting an order

### PR DESCRIPTION
**Pull request by sweir27**
This PR addresses: https://artsyproduct.atlassian.net/browse/PURCHASE-443

We want to handle the following errors (aside from just showing the generic modal):
- `insufficient_inventory`--> show a unique message + link user back to the artist page if they hit "continue"
- `artwork_version_mismatch` --> show a unique message + link user back to the artwork page if they hit "continue"
- `failed_charge_authorize` --> show a unique message (including what we get back from Stripe)

screen shot 2018-09-28 at 9 35 18 am
screen shot 2018-09-28 at 9 36 02 am

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 7f80fe898702423cf5e7231efb7bc2e4a9aca449

https://deploy-preview-1321--artsy-reaction.netlify.com

File: src/Apps/Order/Routes/Payment/index.tsx [hunk 1]
```diff
@@ -263,7 +263,7 @@ export class PaymentRoute extends Component<PaymentProps, PaymentState> {
             this.onMutationError(
               errors || creditCardOrError.mutationError,
               creditCardOrError.mutationError &&
-                creditCardOrError.mutationError.message
+                creditCardOrError.mutationError.detail
```
Comment (sweir27) [inline src/Apps/Order/Routes/Payment/index.tsx]: This fixes an error I saw during the QA session yesterday-- the helpful message from Stripe actually lives in the `detail` in this case.

File: src/Apps/Order/Routes/Review/index.tsx [hunk 2]
```diff
@@ -80,12 +122,39 @@ export class ReviewRoute extends Component<ReviewProps, ReviewState> {
...
   }
 
-  private onMutationError(errors, errorModalMessage?) {
+  private routeToArtworkPage() {
+    const artworkId = get(
+      this.props.order,
+      o => o.lineItems.edges[0].node.artwork.id
+    )
+    // Don't confirm whether or not you want to leave the page
+    this.props.route.onTransition = () => null
```
Comment (sweir27) [inline src/Apps/Order/Routes/Review/index.tsx]: Faffed with this a little bit, but this was the cleanest way I could find to ignore the transition hook behavior just in this case.

File: src/Apps/Order/Routes/Review/index.tsx [hunk 3]
```diff
@@ -66,12 +71,49 @@ export class ReviewRoute extends Component<ReviewProps, ReviewState> {
...
+                submitOrder: { orderOrError },
+              } = result
+
+              const error = orderOrError.error
+              if (error) {
+                switch (error.code) {
+                  case "insufficient_inventory":
+                    const artistId = get(
+                      this.props.order,
+                      o => o.lineItems.edges[0].node.artwork.artist.id
```
Comment (ashkan18) [inline src/Apps/Order/Routes/Review/index.tsx]: what happens when there are multiple artists? https://github.com/artsy/metaphysics/blob/master/src/schema/artwork/index.ts#L78

File: src/Apps/Order/Routes/Review/index.tsx (hunk 2 above)
Comment (zephraph) [reply src/Apps/Order/Routes/Review/index.tsx]: Interesting... We have a cleanup method in OrderApp, but it's not necessarily available here... hmm... 

https://github.com/artsy/reaction/blob/master/src/Apps/Order/OrderApp.tsx#L46

Comment (sweir27) [reply src/Apps/Order/Routes/Review/index.tsx]: Yeah :/ if you think there's a better pattern to adopt let me know. In their example they delegate all handling to the children, but that doesn't seem much better than what we're doing here.

File: src/Apps/Order/Routes/Review/index.tsx (hunk 3 above)
Comment (sweir27) [reply src/Apps/Order/Routes/Review/index.tsx]: We'll have to choose one anyway-- `artist` grabs the first artist when it's returned from gravity, but I can explicitly query for `artists` here and grab the first one client-side to be more explicit.

File: src/Apps/Order/Routes/Review/index.tsx (hunk 2 above)
Comment (zephraph) [reply src/Apps/Order/Routes/Review/index.tsx]: Nope, this is definitely good enough. I don't think it's worth the effort.

**Pull request merged**